src/components/tiptap-extension/highlight/highlight-extension.ts

```diff
@@ -1,16 +1,44 @@
+import { mergeAttributes } from "@tiptap/core"
 import HighlightBase from "@tiptap/extension-highlight"
 
 /**
  * Custom Highlight extension that persists the color attributes.
  */
 export const Highlight = HighlightBase.extend({
+  addAttributes() {
+    if (!this.options.multicolor) {
+      return {}
+    }
+
+    return {
+      color: {
+        default: null,
+        parseHTML: (element) =>
+          element.getAttribute("data-color") || element.style.backgroundColor,
+        renderHTML: (attributes) => {
+          if (!attributes.color) {
+            return {}
+          }
+
+          return {
+            "data-color": attributes.color,
+            style: `background-color: ${attributes.color}; color: inherit`,
+          }
+        },
+      },
+    }
+  },
+
   parseHTML() {
     return [
       {
         tag: "mark",
         getAttrs: (element) => {
           if (!(element instanceof HTMLElement)) return false
 
+          // Exclude underline-highlight marks to avoid conflicts
+          if (element.classList.contains("underline-highlight")) return false
```

Comment:
他のdata-colorを使うextensionが増えたときに全てexcludeしなければいけないので`highlight`クラスがあるかどうかだけ確認してください．

```suggestion
          if (!element.classList.contains("underline")) return false
```